hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketAcceptorThread.java

```diff
@@ -47,8 +49,20 @@
     private final SwCounter eventCount = newSwCounter();
     @Probe
     private final SwCounter exceptionCount = newSwCounter();
+    // count number of times the selector was recreated (if selectWorkaround is enabled)
+    @Probe
+    private final SwCounter selectorRecreateCount = newSwCounter();
+    // last time select returned
     private volatile long lastSelectTimeMs;
 
+    // When true, enables workaround for bug occuring when SelectorImpl.select returns immediately
```

Comment:
I would pull this logic into a method e..g 

selectorRebuildEnabled and add a log statement when it is true.